gotham-server/src/storage/aws/dynamodb.rs

```diff
@@ -0,0 +1,231 @@
+use std::thread;
+use std::time::Duration;
+use std::default::Default;
+use serde_json;
+use serde;
+use rusoto_core::{ProvideAwsCredentials, DispatchSignedRequest};
+use rusoto_dynamodb::*;
+use super::error::*;
+use super::*;
+use std::string::*;
+use std::collections::HashMap;
+use std::fmt::Debug;
+
+#[derive(Serialize, Deserialize)]
+struct DBItem<T> {
+    id: String,
+    v: T,
+}
+
+pub fn insert<T>(dynamodb_client: &DynamoDbClient, id: &str, table_name: &str, v: T) where T: serde::ser::Serialize {
+    let wrapper_item = DBItem {id: id.to_string(), v};
+    let item = serde_dynamodb::to_hashmap(&wrapper_item).unwrap();
+    let put_item_input = PutItemInput {
+        item: item,
+        table_name: table_name.to_string(),
+        ..Default::default()
+    };
+    // Put item
+    let put_item_result = dynamodb_client.put_item(put_item_input).sync();
+}
+
+pub fn get<'a, T>(dynamodb_client: &rusoto_dynamodb::DynamoDbClient, id: &str, table_name: String)
+              -> Option<T> where T: serde::de::Deserialize<'a> {
+    let mut query_key: HashMap<String, AttributeValue> = HashMap::new();
+    query_key.insert(
+        "id".to_string(),
+        AttributeValue {
+            s: Some(id.to_string()),
+            ..Default::default()
+        },
+    );
+    let query_item = GetItemInput {
+        key: query_key,
+        table_name: table_name.to_string(),
+        ..Default::default()
+    };
+
+    match dynamodb_client.get_item(query_item).sync() {
+        Ok(item_from_dynamo) => {
+            match item_from_dynamo.item {
+                None => {
+                    println!("nothing received from Dynamo, item may not exist");
+                    None
+                },
+                Some(attributes_map) => {
+                    let raw_item: serde_dynamodb::error::Result<DBItem<T>> = serde_dynamodb::from_hashmap(attributes_map);
+                    match raw_item {
+                        Ok(s) => {
+                            Some(s.v)
+                        },
+                        Err(e) => {
+                            None
+                        }
+                    }
+                },
+            }
+        },
+        Err(err) => {
+            println!("Error retrieving object: {:?}", err);
```

Comment:
The error here is hidden and can't be consume by a client - What should we do in case this happen?

Reply:
Good point, modified so the DB interface returns `Result` (optionally `Err`) and propagates to the consuming code.